scalr/utils/file_utils.py

```diff
@@ -125,44 +130,49 @@ def write_chunkwise_data(full_data: Union[AnnData, AnnCollection],
         write_data(data, path.join(dirpath, f'{i}.h5ad'))
 
 
-def load_train_val_data_from_config(data_config):
-    """This function returns train & validation data from the data config.
+def _get_datapath_from_config(data_config):
+    """This function returns the datapath to be used to read from config.
 
     Args:
         data_config: Data config.
     """
-
     flow_logger = FlowLogger('File Utils')
-
     train_val_test_paths = data_config.get('train_val_test')
+    datapath = None
+
     if not train_val_test_paths:
         raise ValueError('Split Datapaths not given')
 
     if train_val_test_paths.get('feature_subset_datapaths'):
-        train_data = read_data(
```

Comment:
If we assume user has all 3 in config, then we need to decide the priority. I think
1. final_datapaths
2. feature subsets
2. data splits

If need arises to reuse old compiled config in future, then we need to set if else priorities accordingly.

Reply:
The priority **for reading from config** is set by the if else statements as:
feature_subset > final_datapaths > data_splits.
The `final_datapaths` here means the paths after data ingestion step is complete. 

User config CAN NOT have values in `final_datapaths` or `feature_subset`. If the need arises, the value must be given in `data_split` only. This is the rule for data_ingestion.

Reply:
This means that data by user will only be present in data_splits, and feature_subsetting run hone ke baad only, feature_subset_datapaths can be added, and once added, it is always used by the pipeline later